src/Storage/Mysql/UserRepository.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Bugcache\Storage\Mysql;
+
+use Amp\Mysql\Pool;
+use Amp\Mysql\ResultSet;
+use Amp\Promise;
+use Bugcache\Storage;
+use Generator;
+use function Amp\resolve;
+
+class UserRepository implements Storage\UserRepository {
+    private $mysql;
+
+    public function __construct(Pool $mysql) {
+        $this->mysql = $mysql;
+    }
+
+    public function findByName(string $username): Promise {
+        $sql = "SELECT id, name FROM users WHERE name = ?";
+
+        return resolve($this->fetchUser($sql, [$username]));
+    }
+
+    private function fetchUser(string $sql, array $params = []): Generator {
+        /** @var ResultSet $result */
+        $result = yield $this->mysql->prepare($sql, $params);
```

Comment:
Does a statement prepared at boot time and later on looses connection will work after a reconnect?